grouper/repositories/permission_grant.py

```diff
@@ -51,10 +66,27 @@ def __init__(self, session):
         # type: (Session) -> None
         self.session = session
 
-    def permission_grants_for_user(self, username):
+    def group_grants_for_permission(self, name):
+        # type: (str) -> List[GroupPermissionGrant]
+        permission = Permission.get(self.session, name=name)
+        if not permission or not permission.enabled:
```

Comment:
This was a great point, and it turned into a whole rat hole. Where I ended up at was:

- Service accounts already delete all their permission grants when disabled, so permissions for disabled service accounts aren't a thing, which keeps things simpler
- Groups do keep permission grants when disabled, which allows undoing disabling more easily. I think we eventually want to drop this too, but I'm not ready to do that yet.
- Disabled permissions shouldn't be granted to anything. This is partly because there's no UI to re-enable a disabled permission, so this is more permanent. If the permission is granted to active groups or service accounts, the person disabling it should be forced to remove all of those grants before disabling the permission, since it ensures they've seen all the consequences. But permission grants to disabled groups should just be silently deleted when the permission is disabled, since there's nothing served by keeping them around.

I now have another PR (which depends on this PR and its child since it uses some of the machinery added there) which implements all of that, at which point this code becomes unconditionally correct because there's no way a disabled permission is granted to anything. I'm therefore leaving this as-is, to be cleaned up in that later PR which I'll put up as soon as we get to that point.